pages/index.js

```diff
@@ -1,19 +1,78 @@
 import ProductsList from "@/components/ProductsList";
 import Icon from "@/components/Icons";
 import { StyledCreateLink } from "@/components/Buttons";
+import styled from "styled-components";
+import { useEffect, useState } from "react";
+
+const StyledSearchForm = styled.form`
+  display: flex;
+  margin-block-start: 30px;
+  border-radius: 5px;
+  width: 100%;
+  border: 1px solid;
+  box-shadow: inset 0px 0px 2px var(--shadowColor);
+  padding-inline: 10px;
+  background-color: white;
+  &:focus-within {
+    border: 2px solid var(--accentColor);
+  }
+`;
+
+const StyledSearchInput = styled.input`
+  background: transparent;
+  border: none;
+  outline: none;
+  padding: 10px;
+  width: 100%;
+  font-size: 16px;
+`;
 
 export default function HomePage({
   products,
   onDeleteProduct,
   onToggleShoppingList,
 }) {
+  const [searchTerm, setSearchTerm] = useState("");
+  const [foundProducts, setFoundProducts] = useState();
+  useEffect(() => {
+    setFoundProducts(
+      products.filter((product) =>
+        product.name.toLowerCase().includes(searchTerm.toLowerCase())
+      )
+    );
+  }, [searchTerm, products]);
+
   return (
     <main>
+      <StyledSearchForm>
+        <Icon variant="search" />
+        <StyledSearchInput
+          onChange={(event) => setSearchTerm(event.target.value)}
+          type="text"
+          id="searchTerm"
+          name="searchTerm"
+          placeholder="Search a product"
+        ></StyledSearchInput>
+      </StyledSearchForm>
+      <h2>All Products</h2>
       <ProductsList
-        products={products}
+        products={searchTerm.length === 0 ? products : foundProducts}
         onDeleteProduct={onDeleteProduct}
         onToggleShoppingList={onToggleShoppingList}
       ></ProductsList>
+      {!products && (
+        <>
+          <p>Currently, your product list is empty. </p>
+          <p>You can add products to your list via the the green + button</p>
+        </>
+      )}
+      {searchTerm.length > 0 && foundProducts.length === 0 && (
+        <>
+          <p>{`"${searchTerm}"`} not found. </p>
+          <p>Please check your search input or create a new product. </p>
+        </>
+      )}
```

Comment:
Currently when you are searching and do not have any products in your app all of these paragraphs are displayed which is a lot to read. Reconsider if you want to display all of this. 

You could for example disable the search input field when there are no products available to search for. Just one of many ideas.